core/unit_test/TestMDTeamParallelism.hpp

```diff
@@ -0,0 +1,1666 @@
+                  team, n0, n1, n2, n3);
+
+          Kokkos::parallel_reduce(
+              teamVectorRange,
+              [=](const int& i, const int& j, const int& k, const int& l,
+                  int& vectorSum) { vectorSum += v(leagueRank, i, j, k, l); },
+              teamSum);
+
+          Kokkos::single(Kokkos::PerTeam(team),
+                         [&leagueSum, teamSum]() { leagueSum += teamSum; });
+        },
+        finalSum);
+
+    int expectedSum = get_expected_sum(dims, 5, fillFlattenedIndex);
+
+    EXPECT_EQ(finalSum, expectedSum);
+  }
+
+  template <Kokkos::Iterate OuterDirection = Kokkos::Iterate::Default,
+            Kokkos::Iterate InnerDirection = Kokkos::Iterate::Default>
+  static void test_parallel_reduce_for_6D_MDTeamVectorRange(
+      DimsType const& dims, const int initValue) {
+    using ViewType = typename Kokkos::View<DataType******, ExecSpace>;
+
+    int leagueSize = dims[0];
+    int n0         = dims[1];
+    int n1         = dims[2];
+    int n2         = dims[3];
+    int n3         = dims[4];
+    int n4         = dims[5];
+
+    ViewType v("v", leagueSize, n0, n1, n2, n3, n4);
+    FillConstant fillConstant(initValue);
+    FillFlattenedIndex fillFlattenedIndex(leagueSize, n0, n1, n2, n3, n4);
+
+    Kokkos::parallel_for(
+        Kokkos::MDRangePolicy<ExecSpace, Kokkos::Rank<6>>(
+            {0, 0, 0, 0, 0, 0}, {leagueSize, n0, n1, n2, n3, n4}),
+        KOKKOS_LAMBDA(const int i, const int j, const int k, const int l,
+                      const int m, const int n) {
+          v(i, j, k, l, m, n) = fillFlattenedIndex(i, j, k, l, m, n);
+        });
+
+    int finalSum = 0;
+
+    Kokkos::parallel_reduce(
+        Kokkos::TeamPolicy<ExecSpace>(leagueSize, Kokkos::AUTO),
+        KOKKOS_LAMBDA(TeamType const& team, int& leagueSum) {
+          auto leagueRank = team.league_rank();
+          int teamSum     = 0;
+
+          auto teamVectorRange =
+              Kokkos::MDTeamVectorRange<OuterDirection, InnerDirection>(
+                  team, n0, n1, n2, n3, n4);
+
+          Kokkos::parallel_reduce(
+              teamVectorRange,
+              [=](const int& i, const int& j, const int& k, const int& l,
+                  const int& m, int& vectorSum) {
+                vectorSum += v(leagueRank, i, j, k, l, m);
+              },
+              teamSum);
+
+          Kokkos::single(Kokkos::PerTeam(team),
+                         [&leagueSum, teamSum]() { leagueSum += teamSum; });
+        },
+        finalSum);
+
+    int expectedSum = get_expected_sum(dims, 6, fillFlattenedIndex);
+
+    EXPECT_EQ(finalSum, expectedSum);
+  }
+};
+
+/*--------------------------------------------------------------------------*/
+
+constexpr auto Left  = Kokkos::Iterate::Left;
+constexpr auto Right = Kokkos::Iterate::Right;
+
+TEST(TEST_CATEGORY, MDTeamParallelFor) {
+  using namespace MDTeamParallelism;
+  // int dims[] = {16, 16, 16, 16, 16, 16, 16, 16};
```

Comment:
What's up with this comment? Should it be removed?

Reply:
We saw some unexpected behavior from Kokkos when using that specific dimension size for our test, and at the time, we agreed that it seems to be some separate issue that wasn't caused by any of our changes. We were going to circle back and diagnose the problem again.
But yes, I'll delete this comment and the one below this.